core/src/replay_stage.rs

```diff
@@ -832,6 +863,83 @@ impl ReplayStage {
+                .expect(
+                    "In order for propagated check to fail, \
+                        latest leader must exist in progress map, and thus also in BankForks",
+                )
+                .clone();
+            if last_retransmit_retry_info.slot == bank.slot() {
+                if !progress.is_propagated(bank.slot()) {
+                    warn!(
+                        "retransmit_not_propagated: Not propagated slot:{}",
+                        bank.slot()
+                    );
+                    let backoff = std::cmp::min(
+                        last_retransmit_retry_info.retry_iteration,
+                        RETRANSMIT_BACKOFF_CAP,
+                    );
+                    let time_offset = 2_u64.pow(backoff) * RETRANSMIT_BASE_DELAY_MS;
+                    let elapsed = last_retransmit_retry_info.time.elapsed().as_millis();
+                    if last_retransmit_retry_info.time.elapsed().as_millis() > time_offset.into() {
+                        info!(
+                            "retransmit_not_propagated: retrying retransmit. start_slot:{} bank_slot:{} retry_iteration:{} backoff:{} elapsed:{} time_offset:{}",
+                            start_slot,
+                            bank.slot(),
+                            last_retransmit_retry_info.retry_iteration,
+                            backoff,
+                            elapsed,
+                            time_offset,
+                        );
+                        datapoint_info!(
+                            "replay_stage-retransmit",
+                            ("slot", bank.slot(), i64),
+                            (
+                                "retry_iteration",
+                                last_retransmit_retry_info.retry_iteration,
+                                i64
+                            ),
+                        );
+                        retransmitted_slot = Some(bank.slot());
+                        let _ = retransmit_slots_sender
+                            .send(vec![(bank.slot(), bank.clone())].into_iter().collect());
+                        last_retransmit_retry_info.retry_iteration += 1;
+                        last_retransmit_retry_info.time = Instant::now();
+                    } else {
+                        info!(
+                            "retransmit_not_propagated: bypass retry. elapsed:{} time_offset:{}",
+                            elapsed, time_offset
+                        );
+                    }
+                }
+            } else {
+                last_retransmit_retry_info.slot = bank.slot();
+                last_retransmit_retry_info.time = Instant::now();
+                last_retransmit_retry_info.retry_iteration = 0;
+                info!(
+                    "retransmit_not_propagated: resetting last_retransmit_retry_info {:?}",
+                    &last_retransmit_retry_info
+                );
+            }
+        }
```

Comment:
Note here it's possible we could jump to one fork, then jump back, thereby resetting the `last_retransmit_retry_info` even , even if we have already previously retransmitted the same slot on the old fork we're jumping back to.

For example:

```
        /-----2 (my leader block) -----3 - 6
0 - 1
       \------4 (my leader block) -----5 - 7
```

If Poh is reset from 3 to 5 to 6 to 7, you might reset this over and over. This might be rare, so not a huge deal.

To get around this we could keep around a unique `LastRetransmitInfo` per leader block in a `BTreemap` and purge as we make roots greater than the leader blocks

Reply:
I think this has been addressed by the new entry in the progress map